### PR TITLE
no dagster type on GraphIn/Out

### PR DESCRIPTION
**Pull request by yuhan**
## Summary
follow up on https://github.com/dagster-io/dagster/pull/4624
this PR removes dagster_type on GraphIn and GraphOut and updates the migration guide accordingly.

## Test Plan
bk

## Checklist

- [x] My change requires a change to the documentation and I have updated the documentation accordingly.
  image
- [x] I have added tests to cover my changes.

### CODE REVIEW

File: docs/content-crag/guides/dagster/graph_job_op.mdx [hunk 1]
```diff
@@ -710,7 +710,7 @@ def do_something_else(arg1):
     return arg1
 
 
-@graph(ins={"arg1": GraphIn(int)}, out=GraphOut(int))
+@graph(ins={"arg1": GraphIn()}, out=GraphOut())
```
Comment (sryza) [inline docs/content-crag/guides/dagster/graph_job_op.mdx:713]: Is there any reason to provide a `GraphIn` and `GraphOut` here? They could just be omitted, right?

File: docs/content-crag/guides/dagster/graph_job_op.mdx [hunk 2]
```diff
@@ -726,6 +726,8 @@ def do_it_all():
     do_yet_more(do_two_things(one()))
 ```
 
+Note that `GraphIn` and `GraphOut` no longer take Dagster types, and they now respect the types specified on the corresponding `Op` instead.
```
Comment (sryza) [inline docs/content-crag/guides/dagster/graph_job_op.mdx:729]: I think "underlying" would be slightly clearer than "corresponding"

**Review by sryza — CHANGES_REQUESTED** [review]

File: python_modules/dagster/dagster_tests/core_tests/graph_tests/test_graph_ins_out.py [hunk 3]
```diff
@@ -62,32 +62,32 @@ def my_graph():
     }
 
 
-def test_ins_fail():
-    with pytest.raises(DagsterInvalidDefinitionError):
+# def test_ins_fail():
```
Comment (sryza) [inline python_modules/dagster/dagster_tests/core_tests/graph_tests/test_graph_ins_out.py:65]: Should this be deleted?

File: docs/content-crag/guides/dagster/graph_job_op.mdx (hunk 1 above)
Comment (yuhan) [reply docs/content-crag/guides/dagster/graph_job_op.mdx]: Right. updated the GraphIn part.

But we can't omit GraphOut bc we can't easily tell if that's a multi out case at run time. According to the docstring, this is the similar case in `@op` where we also need to do `@op(out={'word': Out(), 'num': Out()})` - not sure if we plan to omit that on op too or have already done so. cc @dpeng817

**Review by sryza — APPROVED** [review]

**Comment by sryza** [comment]
One more thing: GraphIn and GraphOut need to be added to the top-level API

**Pull request merged**